### DISCUSSION TITLE
What means plain type and bounded type？

### DISCUSSION THREAD

**Discussion by wjhtinger** [eProsima/Fast-DDS#3806] [Q&A]
What differece, Can give some example?

**Comment by JLBuenoLopez** [comment]
Hi @wjhtinger 

A bounded type is that which has a limit. For instance, IDL syntax allows to define a sequence, string and map without specifying the maximum number of elements of the container. Thus, the type is considered unbounded if the optional upper limit parameter is not defined.

A plain type is a type whose CDR representation matches its in-memory representation. This requirement avoids the copy between the CDR buffer and the user buffer because the data representation is the same. Consequently, only primitive types are considered to be plain (Fast DDS also provides an API to check if a defined type is plain: [`TypeSupport::is_plain`](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/dds_pim/topic/typesupport.html#_CPPv4NK8eprosima7fastdds3dds11TypeSupport8is_plainEv)). That is the reason why the zero-copy example defines `char message[256]` instead of `string<256> message`. String is not a plain type and zero-copy mechanism would not work with that data type.

**Reply by wjhtinger** [reply]
If I want to use string, then how to limit it to make it bounded?
And if set is_bounded() return true, but the string memory grow dynamically, what will happen?
